boards/PDM/Src/Io/Io_Efuse.c

```diff
@@ -0,0 +1,346 @@
+    SPI_HandleTypeDef *const spi_handle,
+    GPIO_TypeDef *           nss_port,
+    uint16_t                 nss_pin,
+    GPIO_TypeDef *           fsob_port,
+    uint16_t                 fsob_pin,
+    GPIO_TypeDef *           fsb_port,
+    uint16_t                 fsb_pin,
+    GPIO_TypeDef *           channel0_port,
+    uint16_t                 channel0_pin,
+    GPIO_TypeDef *           channel1_port,
+    uint16_t                 channel1_pin)
+{
+    assert(spi_handle != NULL);
+
+    struct Efuse_Context *efuse_context = malloc(sizeof(struct Efuse_Context));
+    assert(efuse_context != NULL);
+
+    efuse_context->hspi            = spi_handle;
+    efuse_context->nss_port        = nss_port;
+    efuse_context->nss_pin         = nss_pin;
+    efuse_context->fsob_port       = fsob_port;
+    efuse_context->fsob_pin        = fsob_pin;
+    efuse_context->fsb_port        = fsb_port;
+    efuse_context->fsb_pin         = fsb_pin;
+    efuse_context->channel0_port   = channel0_port;
+    efuse_context->channel0_pin    = channel0_pin;
+    efuse_context->channel1_port   = channel1_port;
+    efuse_context->channel1_pin    = channel1_pin;
+    efuse_context->wdin_bit_to_set = true;
+
+    return efuse_context;
+}
+
+void Io_Efuse_Destroy(struct Efuse_Context *efuse)
+{
+    free(efuse);
+}
+
+void Io_Efuse_EnableChannel0(const struct Efuse_Context *const efuse)
+{
+    HAL_GPIO_WritePin(efuse->channel0_port, efuse->channel0_pin, GPIO_PIN_SET);
+}
+
+void Io_Efuse_DisableChannel0(const struct Efuse_Context *const efuse)
+{
+    HAL_GPIO_WritePin(
+        efuse->channel0_port, efuse->channel0_pin, GPIO_PIN_RESET);
+}
+
+void Io_Efuse_EnableChannel1(const struct Efuse_Context *const efuse)
+{
+    HAL_GPIO_WritePin(efuse->channel1_port, efuse->channel1_pin, GPIO_PIN_SET);
+}
+
+void Io_Efuse_DisableChannel1(const struct Efuse_Context *const efuse)
+{
+    HAL_GPIO_WritePin(
+        efuse->channel1_port, efuse->channel1_pin, GPIO_PIN_RESET);
+}
+
+ExitCode Io_Efuse_ConfigureChannelMonitoring(
+    uint8_t                     monitoring_function,
+    struct Efuse_Context *const efuse)
+{
+    uint16_t register_value;
+
+    // Read original content of GCR Register
+    RETURN_CODE_IF_EXIT_NOT_OK(
+        Io_Efuse_ReadRegister(SI_GCR_ADDR, &register_value, efuse));
+
+    CLEAR_BIT(register_value, (CSNS1_EN_MASK | CSNS0_EN_MASK));
+    SET_BIT(
+        register_value,
+        (monitoring_function & (CSNS1_EN_MASK | CSNS0_EN_MASK)));
+
+    RETURN_CODE_IF_EXIT_NOT_OK(
+        Io_Efuse_WriteRegister(SI_GCR_ADDR, register_value, efuse));
+
+    return EXIT_CODE_OK;
+}
+
+ExitCode Io_Efuse_ExitFailSafeMode(struct Efuse_Context *const efuse)
+{
+    // Set the WDIN bit (15th bit) to exit out of fail-safe mode, the parity bit
+    // (14th bit) will automatically be set by the Io_Efuse_WriteRegister()
+    // function 1_1_00000_00000_0000
+    efuse->wdin_bit_to_set = true;
+
+    RETURN_CODE_IF_EXIT_NOT_OK(
```

Comment:
Why are we sending 0's here?

Reply:
We send 0x000 into the Io_Efuse_WriteRegister() and since efuse->wdin_bit_to_set = true; was set in the line before the data that gets sent over SPI is
11_000000_00000000

Reply:
> Why are we sending 0's here? Magic numbers moment

😠